A522302143163396 auction parameters

Scandie opened the pull request.

Scandie committed:
Add setting dutch rounds amount from API

Scandie committed:
Add test for setting dutch rounds amount from API

- Update data for unit and functional tests with new field 'auctionParameters'

yarsanich commented on openprocurement/auction/insider/mixins.py:
```diff
@@ -89,6 +89,7 @@ def get_auction_info(self, prepare=False):
             for bid in self._auction_data['data'].get('bids', [])
             if bid.get('status', 'active') == 'active'
         ]
+        self.dutch_rounds = self._auction_data['data'].get('auctionParameters', {}).get('steps', DUTCH_ROUNDS - 1) + 1
```
На рівні `Auction` класу, який знаходиться в `auction.py` модулі, винести `dict` параметрів, `parameters` і дати докстрінг, що ці параметри приходять з API.

yarsanich commented:
Потрібно ще

- Додати параметри аукціону в документ в базі
- Додати параметри аукціону в аудит.

Scandie committed:
Add attribute parameters to Auction class, which contains auctionParameters from API

- Add auctionParameters to auction document
- Add auctionParameters to audit document
- Update tests

yarsanich commented on openprocurement/auction/insider/utils.py:
```diff
@@ -261,14 +262,16 @@ def prepare_auction_document(auction, fast_forward=False):
         from openprocurement.auction.insider.constants import DUTCH_TIMEDELTA,\
-            DUTCH_ROUNDS, FIRST_PAUSE
+            FIRST_PAUSE
+        DUTCH_ROUNDS = auction.parameters['steps'] + 1
```
docsting сюди, навіщо потрібно +1

Scandie committed:
Add coment to 'prepare_auction_document' util function

yshalenyk commented on openprocurement/auction/insider/utils.py:
```diff
@@ -261,14 +262,17 @@ def prepare_auction_document(auction, fast_forward=False):
-            DUTCH_ROUNDS, FIRST_PAUSE
+            FIRST_PAUSE
+        # Additional round is needed to provide set amount of steps (auction starts with initial price)
+        DUTCH_ROUNDS = auction.parameters['steps'] + 1
```
якщо це з api приходить я би тут explicit int(auction.parameters['steps']) зробив

Scandie committed:
Add explicit auction.parameters['steps'] converting to int

Scandie committed:
Rename auctionParameters.steps to auctionParameters.dutchSteps

- Change default auctionParameters.type to 'insider'
- Update tests

yarsanich commented on openprocurement/auction/insider/tests/unit/test_utils.py:
```diff
@@ -91,6 +92,7 @@ def test_prepare_audit(auction, mocker):
         }
     }
 
+    auction.get_auction_info()
```
А нащо тобі брати дані?
